controllers/storagecluster/cephcluster.go

```diff
@@ -1148,16 +1148,72 @@ func createPrometheusRules(r *StorageClusterReconciler, sc *ocsv1.StorageCluster
 		return err
 	}
 	applyLabels(getCephClusterMonitoringLabels(*sc), &prometheusRule.ObjectMeta)
-	replaceTokens := []exprReplaceToken{
+
+	replaceTokens := []replaceToken{
 		{
 			recordOrAlertName: "CephMgrIsAbsent",
-			wordInExpr:        "openshift-storage",
+			wordToReplace:     "openshift-storage",
 			replaceWith:       sc.Namespace,
 		},
 	}
+
+	// if different than default nearFullRatio or fullRatio values are provided, replace the values in the prometheus rule accordingly
+	if sc.Spec.ManagedResources.CephCluster.NearFullRatio != nil || sc.Spec.ManagedResources.CephCluster.FullRatio != nil {
+		//
+		var (
+			nearFullRatio = 0.75
+			fullRatio     = 0.85
+		)
+		if sc.Spec.ManagedResources.CephCluster.NearFullRatio != nil {
+			nearFullRatio = *sc.Spec.ManagedResources.CephCluster.NearFullRatio
+		}
+		if sc.Spec.ManagedResources.CephCluster.FullRatio != nil {
+			fullRatio = *sc.Spec.ManagedResources.CephCluster.FullRatio
+		}
+
+		criticallyFullRatio := (nearFullRatio + fullRatio) / 2
+
+		if nearFullRatio != 0.75 {
+			replaceTokens = append(replaceTokens,
+				replaceToken{
+					wordToReplace: "75%",
+					replaceWith:   fmt.Sprintf("%.2f%%", nearFullRatio*100),
+				},
+				replaceToken{
+					wordToReplace: "0.75",
+					replaceWith:   fmt.Sprintf("%f", nearFullRatio),
+				},
+			)
```

Comment:
How about we create a func and pass these values to it? It will reduce the duplication.

Reply:
Thanks for the suggestion, Done.